pkg/java/jar.go

```diff
@@ -96,25 +112,34 @@ func md5File(file fs.File) (string, int64, error) {
 }
 
 func md5Bytecode(r *zip.ReadCloser, classLocation string) (string, error) {
-	c, err := r.Open(classLocation + ".class")
+	buf, err := readClassBytes(r, classLocation)
 	if err != nil {
 		return "", err
 	}
 
-	buf := new(bytes.Buffer)
-	_, err = buf.ReadFrom(c)
+	h, err := HashClassInstructions(buf.Bytes())
```

Comment:
nit: can we just return `HashClassInstructions(buf.Bytes())` here? I feel that if there's an error being returned, the caller shouldn't be using the `string` returned anyway.